gtfs-validator/main/src/main/java/org/mobilitydata/gtfsvalidator/validator/StopTooFarFromTripShapeValidator.java

```diff
@@ -0,0 +1,131 @@
+        // Cache for previously tested shape_id and stop_id pairs - no need to test them more than once.
+        final Map<String, List<String>> testedCache = new HashMap<>();
+        // Go through the pair of tripId and the corresponding stop time list one by one.
+        for (Map.Entry<String, List<GtfsStopTime>> tripIdStopTimeListEntry : Multimaps.asMap(stopTimeTable.byTripIdMap()).entrySet()) {
+            final String tripId = tripIdStopTimeListEntry.getKey();
+            final List<GtfsStopTime> stopTimeList = tripIdStopTimeListEntry.getValue();
+            final GtfsTrip trip = tripTable.byTripId(tripId);
+            if (trip == null || stopTimeList == null || stopTimeList.isEmpty() || trip.shapeId() == null) {
+                // This rule only applies when all necessary fields exist.
+                continue;
+            }
+            final List<GtfsShape> shapeList = shapeTable.byShapeId(trip.shapeId());
+            if (shapeList == null || shapeList.isEmpty()) {
+                continue;
+            }
+            // Get the trip shape polygon given the Gtfs shapes data and TRIP_BUFFER_DEGREES.
+            Shape tripShapeGivenThreshold = createTripShapePolygonGivenThreshold(shapeList);
+            // Check if each stop is within the buffer polygon.
+            for (GtfsStopTime stopTime : stopTimeList) {
+                GtfsStop stop = stopTable.byStopId(stopTime.stopId());
+                if (stop == null) {
+                    continue;
+                }
+                if (!(stop.locationType() == GtfsLocationType.STOP) && !(stop.locationType() == GtfsLocationType.BOARDING_AREA)) {
+                    // This rule only applies to stops of location_type "STOP" or "BOARDING_AREA".
+                    continue;
+                }
+                if (!testedCache.containsKey(trip.shapeId())) {
+                    // Record the shape_id and stop_id pair when shape_id never appeared before.
+                    List<String> stopIds = new ArrayList<>();
+                    stopIds.add(stop.stopId());
+                    testedCache.put(trip.shapeId(), stopIds);
+                } else if (testedCache.get(trip.shapeId()).contains(stop.stopId())) {
+                    // Skip tested shape_id and stop_id pair.
+                    continue;
+                } else {
+                    // Record the new shape_id and stop_id pair when shape_id has appeared.
+                    testedCache.get(trip.shapeId()).add(stop.stopId());
+                }
+                // Check whether the stop position is within an acceptable distance threshold from the trip shape.
+                Point p = getShapeFactory().pointXY(stop.stopLon(), stop.stopLat());
+                if (!tripShapeGivenThreshold.relate(p).equals(SpatialRelation.CONTAINS)) {
+                    noticeContainer.addNotice(new StopTooFarFromTripShapeNotice(stop.stopId(), stopTime.stopSequence(), trip.tripId(), trip.shapeId(), TRIP_BUFFER_METERS));
```

Comment:
Nit: please fix formatting

Reply:
I have adjusted the format referring to the clang format result, while still keeping it 4 space be to consistent with other rules' files.

Reply:
I have run google-java-format to fix formatting again.